src/buffer/out/TextAttribute.cpp

```diff
@@ -81,6 +81,32 @@ bool TextAttribute::IsLegacy() const noexcept
     return _foreground.IsLegacy() && _background.IsLegacy();
 }
 
+// Routine Description:
+// - Makes this TextAttribute's foreground color the same as the other one.
+// Arguments:
+// - The TextAttribute to copy the foreground color from
+// Return Value:
+// - <none>
+void TextAttribute::SetForegroundFrom(const TextAttribute& other) noexcept
+{
+    _foreground = other._foreground;
+    WI_ClearAllFlags(_wAttrLegacy, FG_ATTRS);
+    _wAttrLegacy |= (other._wAttrLegacy & FG_ATTRS);
+}

```

Comment:
I'm fairly certain you don't need these methods. The `_wAttrLegacy` field is only used to store meta attributes, so there's no need to replicate the color attribute bits here. So really all you should need are the existing `SetForeground` and `SetBackground` methods.

Reply:
Yay! Thanks James!